hardware_interface/include/hardware_interface/internal/interface_manager.h

```diff
@@ -142,12 +162,18 @@ class InterfaceManager
   /**
    * \brief Get an interface.
    *
-   * Since this class only stores one interface per type, this returns a
-   * pointer to the requested interface type. If the interface type is not
-   * registered, it will return \c NULL.
+   * If this class and its registered sub-managers only have one registered
+   * instance of the requested type, this will be returned. If multiple
+   * instances of the interface type were registered and the type is a
+   * \c ResourceManager, this call will try to combine them into a single
+   * handle. In all other cases, \c nullptr will be returned.
+   *
+   * \note As this instance will only store one instance of each manager type,
+   * the only way multiple interfaces are registered is the registration of
+   * other managers with interfaces of the same type.
```

Comment:
Please add a note to this explaining that, in this case, this function will return null because there is no method of combining these non-`ResourceManager` interfaces.

I just want to make sure it's very clear in the documentation that this function has a shortcoming under that condition, since it seems to me that that condition (multiple interfaces of the same type which isn't `ResourceManager`) is quite possible. To keep API compatible, returning nullptr is the best we can do, but it does seem like a shortcoming over a more robust solution like returning multiple interfaces or returning the directly-managed interface over sub-manager interfaces or something like that.

Reply:
Added in rebased commits